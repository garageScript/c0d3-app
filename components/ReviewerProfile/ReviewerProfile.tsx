import React from 'react'
import styles from './reviewerProfile.module.scss'

type ReviewerProfileProps = {
  username: string | undefined | null
  name: string | undefined | null
  inline?: boolean
}
const ReviewerProfile: React.FC<ReviewerProfileProps> = ({
  username,
  name,
  inline
}) => {
  //TO-DO fix User type to make these fields non-nullable
  const firstName = name ? name.split(' ')[0] : ''
  const lastName = name?.split(' ')[1] || ''
  return (
    <div className={`${inline ? styles['wrapper'] : ''}`}>
      <a
        className={`${
          styles[inline ? 'comment_author__inline' : 'comment_author']
        } mt-2 `}
        href={username ? `/profile/${username}` : undefined}
      >
        <div className="d-inline">{`${firstName} ${lastName}`}</div>
        <div className="d-inline text-muted">
          {username ? ' @' + username : ''}
        </div>
      </a>
    </div>
  )
}
export default ReviewerProfile
