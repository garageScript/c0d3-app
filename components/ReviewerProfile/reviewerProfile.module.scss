.comment_author__inline {
  margin-left: 1em;
  font-size: 0.85rem;
  color: gray;
  width: max-content;
  &:hover {
    color: black;
    text-decoration-color: gray;
  }
}
.comment_author {
  color: black;
  width: max-content;
  &:hover {
    color: black;
    text-decoration-color: gray;
  }
}
.wrapper {
  background-color: #fafbfc;
}
