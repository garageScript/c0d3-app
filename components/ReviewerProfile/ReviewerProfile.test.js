import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import ReviewerProfile from './ReviewerProfile'

describe('ReviewerProfile test', () => {
  test('Should render default component', () => {
    const { container } = render(
      <ReviewerProfile name="User User" username="user" />
    )
    expect(screen.getByText('@user')).toBeVisible()
    expect(container).toMatchSnapshot()
  })
  test('Should use inline comment if props is passed', () => {
    const { container } = render(
      <ReviewerProfile name="User User" username="user" inline />
    )
    expect(screen.getByText('@user')).toBeVisible()
    expect(container.firstChild.classList.contains('wrapper')).toBe(true)
  })
})
