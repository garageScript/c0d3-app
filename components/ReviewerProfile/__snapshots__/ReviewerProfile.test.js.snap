// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewerProfile test Should render default component 1`] = `
<div>
  <div
    class=""
  >
    <a
      class="comment_author mt-2 "
      href="/profile/user"
    >
      <div
        class="d-inline"
      >
        User User
      </div>
      <div
        class="d-inline text-muted"
      >
         @user
      </div>
    </a>
  </div>
</div>
`;
